pkg/controller/syncset/syncset_controller_test.go

```diff
@@ -0,0 +1,313 @@
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var (
+	configMapGVK = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ConfigMap"}
+	nsGVK        = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Namespace"}
+	podGVK       = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"}
+)
+
+const (
+	timeout = time.Second * 10
+	tick    = time.Second * 2
+)
+
+// Test_ReconcileSyncSet_wConfigController verifies that SyncSet and Config resources
+// can get reconciled and their respective specs are added to the data client.
+func Test_ReconcileSyncSet_wConfigController(t *testing.T) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	require.NoError(t, testutils.CreateGatekeeperNamespace(cfg))
+
+	tr := setupTest(ctx, t, false, true)
+	mgr := *tr.mgr
+	c := tr.c
+	cfClient := tr.cfClient
+
+	instanceConfig := testutils.ConfigFor([]schema.GroupVersionKind{})
+	instanceSyncSet1 := &syncsetv1alpha1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "syncset1",
+		},
+	}
+	instanceSyncSet2 := &syncsetv1alpha1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "syncset2",
+		},
+	}
+	configMap := testutils.UnstructuredFor(configMapGVK, "", "cm1-name")
+	pod := testutils.UnstructuredFor(podGVK, "", "pod1-name")
+
+	// for sync controller
+	syncAdder := syncc.Adder{CacheManager: tr.cacheMgr, Events: tr.events}
+	require.NoError(t, syncAdder.Add(mgr), "adding sync reconciler to mgr")
+
+	// now for config controller
+	configAdder := config.Adder{
+		CacheManager:     tr.cacheMgr,
+		ControllerSwitch: tr.cs,
+		Tracker:          tr.tracker,
+	}
+	require.NoError(t, configAdder.Add(mgr), "adding config reconciler to mgr")
+
+	testutils.StartManager(ctx, t, mgr)
+
+	require.NoError(t, c.Create(ctx, configMap), fmt.Sprintf("creating ConfigMap %s", "cm1-mame"))
+	require.NoError(t, c.Create(ctx, pod), fmt.Sprintf("creating Pod %s", "pod1-name"))
+
+	tts := []struct {
+		name         string
+		setup        func(t *testing.T)
+		cleanup      func(t *testing.T)
+		expectedGVKs []schema.GroupVersionKind
+	}{
+		{
+			name: "config and 1 sync",
+			setup: func(t *testing.T) {
+				t.Helper()
+
+				instanceConfig := testutils.ConfigFor([]schema.GroupVersionKind{configMapGVK, nsGVK})
+				instanceSyncSet := &syncsetv1alpha1.SyncSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "syncset1",
+					},
+					Spec: syncsetv1alpha1.SyncSetSpec{
+						GVKs: []syncsetv1alpha1.GVKEntry{
+							syncsetv1alpha1.GVKEntry(podGVK),
+						},
+					},
+				}
+
+				require.NoError(t, c.Create(ctx, instanceConfig))
+				require.NoError(t, c.Create(ctx, instanceSyncSet))
+			},
+			cleanup: func(t *testing.T) {
```

Comment:
Do we need per-testcase cleanup?  Why not just clean everything after each test case and put it in `t.Run()`?